Carbon.Cryptography/Functions/Convert-CCertificateProvider.ps1

```diff
@@ -0,0 +1,161 @@
+    Converts the provider of a certificate's private key.
+
+    .DESCRIPTION
+    The `Convert-CCertificateProvider` function changes the provider of a certificate's private key. Pass the path to
+    the certificate file to the `FilePath` parameter, and the new provider name to the `ProviderName` parameter. If the
+    certificate file is password-protected, pass the password to the `Password` parameter.
+
+    If the private key's provider is already the value passed to the function, nothing happens.
+
+    The function uses the `certutil` command to import the certificate with its private key into a "Temp" store for the
+    current user using the new provider. This command actually does the conversion process. Then,
+    `Convert-CCertificateProvider` exports the certificate, overwriting the original file. (If the `Password` parameter
+    has a value, the certificate file is password-protected with that password.) The temporary certificate is removed
+    from the current user's "Temp" store. Finally, the function returns an object with the following properties:
+
+    * `Path`: the path to the file that was converted
+    * `OldProviderName`: the name of the private key's original/old provider name
+    * `NewProviderName`: the name of the private key's new provider
+    * `NewCertificateBase64Encoded`: the raw bytes of the new certificate file, base-64 encoded.
+
+    The `certutil -csplist` shows a list of available cryptographic providers.
+
+    .EXAMPLE
+    Convert-CCertificateProvider -FilePath .\mycert.pfx -ProviderName 'Microsoft Enhanced RSA and AES Cryptographic Provider'
+
+    Demonstrates how to convert the provider of a certificate's private key and the certificate file is ***not***
+    password protected.
+
+    .EXAMPLE
+    Convert-CCertificateProvider -FilePath .\mycert.pfx -ProviderName 'Microsoft Enhanced RSA and AES Cryptographic Provider'
+
+    Demonstrates how to convert the provider of a certificate's private key and the certificate file ***is*** password
+    protected.
+    #>
+    [CmdletBinding()]
+    param(
+        # The path to the certificate file to convert. Must have a private key.
+        [Parameter(Mandatory)]
+        [String] $FilePath,
+
+        # The new provider name for the certifcate's private key. The `certutil -csplist` command shows the list of
+        # available cryptographic providers.
+        [Parameter(Mandatory)]
+        [String] $ProviderName,
+
+        # The password for the certificate file, if any. When replacing the existing certificate file, it will be
+        # protected with the same password (or not protected if no password is passed).
+        [securestring] $Password
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    if (-not (Test-Path -Path $FilePath -PathType Leaf))
+    {
+        "Unable to convert provider for certificate ""$($FilePath)"" because it does not exist." |

```

Comment:
This error message is confusing, it almost sounds like the provider does not exist. I'd make it more explicit that it's the given file path that does not exist.